src/PackageStorage/Application/Metric/ModuleStatisticUpdater.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+declare(strict_types=1);
+
+namespace PackageStorage\Application\Metric;
+
+use PackageStorage\Application\Fetcher\ProjectExtendedClassesFetcherInterface;
+use PackageStorage\Application\Fetcher\VendorChangedClassesFetcherInterface;
+use Upgrade\Application\Dto\StepsResponseDto;
+
+class ModuleStatisticUpdater implements ModuleStatisticUpdaterInterface
+{
+    /**
+     * @var \PackageStorage\Application\Fetcher\ProjectExtendedClassesFetcherInterface
+     */
+    protected ProjectExtendedClassesFetcherInterface $projectExtendedClassesFetcher;
+
+    /**
+     * @var \PackageStorage\Application\Fetcher\VendorChangedClassesFetcherInterface
+     */
+    protected VendorChangedClassesFetcherInterface $vendorChangedClassesFetcher;
+
+    /**
+     * @param \PackageStorage\Application\Fetcher\ProjectExtendedClassesFetcherInterface $projectExtendedClassesFetcher
+     * @param \PackageStorage\Application\Fetcher\VendorChangedClassesFetcherInterface $vendorChangedClassesFetcher
+     */
+    public function __construct(
+        ProjectExtendedClassesFetcherInterface $projectExtendedClassesFetcher,
+        VendorChangedClassesFetcherInterface $vendorChangedClassesFetcher
+    ) {
+        $this->projectExtendedClassesFetcher = $projectExtendedClassesFetcher;
+        $this->vendorChangedClassesFetcher = $vendorChangedClassesFetcher;
+    }
+
+    /**
+     * @param \Upgrade\Application\Dto\StepsResponseDto $stepsResponseDto
+     *
+     * @return \Upgrade\Application\Dto\StepsResponseDto
+     */
+    public function updateStatisticPreRequire(StepsResponseDto $stepsResponseDto): StepsResponseDto
+    {
+        $totalOverwritedModels = count($this->projectExtendedClassesFetcher->fetchExtendedClasses());
+        $stepsResponseDto->getModelStatisticDto()->setTotalOverwrittenModels($totalOverwritedModels);
+
+        return $stepsResponseDto;
+    }
+
+    /**
+     * @param \Upgrade\Application\Dto\StepsResponseDto $stepsResponseDto
+     *
+     * @return \Upgrade\Application\Dto\StepsResponseDto
+     */
+    public function updateStatisticPostRequire(StepsResponseDto $stepsResponseDto): StepsResponseDto
+    {
+        $totalChangedCoreModules = $this->vendorChangedClassesFetcher->fetchVendorChangedClassesWithPackage();
+        $projectExtendedClasses = $this->projectExtendedClassesFetcher->fetchExtendedClasses();
+        $stepsResponseDto->getModelStatisticDto()->setTotalChangedModels(count($totalChangedCoreModules));
+
+        $totalIntersectingModels = 0;
+        foreach ($projectExtendedClasses as $projectExtendedClass => $fileName) {
```

Comment:
Maybe this could be used instead https://www.php.net/manual/en/function.array-intersect-key.php + count()?